UpdateSource/src/main/java/update/Updater.java

```diff
@@ -9,101 +9,94 @@
 import java.time.LocalDate;
 import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
 import java.util.stream.Collectors;
+import javax.sql.DataSource;
 import update.Source.CBRFSource;
+import update.Source.CurrencyExchangeRateSource;
 import update.dto.CurrencyDto;
 import update.dto.ExchageRateDto;
 
 public class Updater {
 
-  private CBRFSource cbrfSource;
+  private CurrencyExchangeRateSource source;
   private CurrenciesRepository currenciesRepository;
 
   private ExchangeRateRepository exchangeRateRepository;
 
-  private List<Integer> listCurrencyId;
-
-  private HashMap<Integer, CurrencyDto> hashMap;
+  private HashMap<Integer, CurrencyDto> currencyDtoHashMap;
 
+  private LocalDate date;
 
   private ExchageRateDto exchageRateDto;
 
-  public Updater() {
-
-    this.currenciesRepository = new CurrenciesRepository(null);
-    this.exchangeRateRepository = new ExchangeRateRepository(null);
-    this.cbrfSource = new CBRFSource();
+  public Updater(DataSource dataSource, CurrencyExchangeRateSource source, LocalDate date) {
 
+    this.currenciesRepository = new CurrenciesRepository(dataSource);
+    this.exchangeRateRepository = new ExchangeRateRepository(dataSource);
+    this.source = source;
+    this.date = date;
   }
 
   private void update() {
-    exchageRateDto = cbrfSource.get(LocalDate.now());
-    listCurrencyId = new ArrayList<>();
-//    updateCurrency(exchageRateDto);
-//    updateExchageRate();
+    exchageRateDto = source.get(date);
 
   }
 
   public void updateExchageRate() {
 
+    exchageRateDto = source.get(date);
 
-    // Массив charcode
-    List<String> arrayCharCode = exchageRateDto.getCurrencies().stream()
-        .map(CurrencyDto::getCharCode)
-        .collect(Collectors.toList());
-    // Массив integer  и хэшмап id -> объект ЦБ
-    List<Integer> currencies = new ArrayList<>();
-    for (String charCode : arrayCharCode) {
-      int idCurrency = currenciesRepository.findCodeId(charCode);
-      if (idCurrency != -1) {
-        currencies.add(idCurrency);
-        hashMap.put(idCurrency, getCurrencyByCharCode(exchageRateDto, charCode));
-      } else {
-        //create
-      }
+    // 1. Generate a charCode array
+    List<String> arrayCharCode = new ArrayList<>();
+    for (CurrencyDto curency : exchageRateDto.getCurrencies()) {
+      arrayCharCode.add(curency.getCharCode());
     }
 
-    // Из базы извлекаешь по всем полученным id курсы валют за дату
-    var targetRate = exchangeRateRepository.findByDate(id, Date.valueOf(exchageRateDto.getDate()));
+    // 2. For this array, request currency identifiers from the database
+    HashMap<Integer, String> idsDataBase = currenciesRepository.currencuIds(arrayCharCode);
```

Comment:
```suggestion
    HashMap<Integer, String> idsDataBase = currenciesRepository.currencuIds(charCodeMap.keySet());
```

currencyIds - должен поддерживать интерфейс Collection или Iterable